mage_ai/server/websocket.py

```diff
@@ -183,6 +183,10 @@ def on_message(self, raw_message):
             # Need to use Python magic command for changing directories
             code = re.sub(r'^!%cd|^!cd', '%cd', code)
 
+            install_packages = message.get('install_packages')
+            if install_packages:
+                code = f'!pip install -r {get_repo_path()}/requirements.txt'
```

Comment:
@wangxiaoyou1993 Does using a flag in the request message work? Or is there a better way?

Reply:
It doesn't work. `get_repo_path()` can't be used in bash command. You can get repo_path from backend in one API. For example, include the repo_path in `/api/status` https://github.com/mage-ai/mage-ai/blob/master/mage_ai/server/server.py#L417, and then use that in the command.